veracruz-utils/src/policy/principal.rs

```diff
@@ -14,26 +14,136 @@
 
 use serde::{Deserialize, Serialize};
 use super::error::PolicyError;
-use std::{string::String, vec::Vec};
+use std::{string::String, vec::Vec, collections::{HashMap, HashSet}};
 
 ////////////////////////////////////////////////////////////////////////////////
-// Roles.
+// File operation and capabilities.
 ////////////////////////////////////////////////////////////////////////////////
 
-/// Defines the role (or mix of roles) that a principal can take on in any
-/// Veracruz computation.
+/// List of file operations
+/// TODO: line up  wasi operations eps. the `Right` defined in wasi.
```

Comment:
Is this TODO relevant?

Reply:
This is a tag for wasi branch. I intent to keep it.